lib/ProxySQL_Admin.cpp

```diff
@@ -7029,6 +7031,7 @@ char * ProxySQL_Admin::get_variable(char *name) {
 		sprintf(intbuf, "%d", variables.p_memory_metrics_interval);
 		return strdup(intbuf);
 	}
+	if (!strcasecmp(name,"tls_version")) return s_strdup(variables.tls_version);
```

Comment:
It's used in `ProxySQL_Admin::get_variable` for the same purpose in multiple places:

- https://github.com/sysown/proxysql/blob/v2.1.1/lib/ProxySQL_Admin.cpp#L7035
- https://github.com/sysown/proxysql/blob/v2.1.1/lib/ProxySQL_Admin.cpp#L7036
- https://github.com/sysown/proxysql/blob/v2.1.1/lib/ProxySQL_Admin.cpp#L7037
- https://github.com/sysown/proxysql/blob/v2.1.1/lib/ProxySQL_Admin.cpp#L7040
- https://github.com/sysown/proxysql/blob/v2.1.1/lib/ProxySQL_Admin.cpp#L7066

Probably because it performs and extra check on the passed pointer to check for `nullity` before calling the actual `strdup`.